NorthstarDLL/core/sourceinterface.h

```diff
@@ -1,6 +1,13 @@
 #pragma once
 #include <string>
 
+// interface return status
+enum class InterfaceStatus : int
```

Comment:
this should maybe have a deprecated option

Reply:
The enum is copied from source. It's honestly pretty useless because you get a nullpointer when it fails anyways.
I don't really see the value of providing a status for deprecated interfaces. What is the plugin going to do when it's deprecated?
Storing if an interface is deprecated also adds a bit more complexity.

Reply:
the interface status can give more a reasons for why you would get a null pointer or just as a notice for deprecation

Reply:
But what are plugins going to do about it?
Old interfaces work the exact same as newer ones, they just have less functionality because new functions get added at the bottom of the vftable.
When an interface is removed, the plugin should realize that it got a null pointer and the status IFACE_FAILED and unload itself / shutdown

Reply:
> But what are plugins going to do about it?

better feedback to the end users.

Reply:
there isn't any feedback needed because old interfaces work the same like newer ones, they just have less functions available.
There are no deprecated interfaces. At most, the individual functions of the interface should be deprecated and in that case they should log a warning or something themselves and not the consuming plugin.